### PR TITLE
Suggested changes to `kneth/realm-core-14.0.1`

### PR DESCRIPTION
**Pull request by kraenhansen**
## What, How & Why?

Instead of a lot of suggested changes in the GitHub PR UI, I thought a PR would be better.

### CODE REVIEW

File: packages/realm/src/Logger.ts [hunk 1]
```diff
@@ -146,7 +146,7 @@ export enum LogCategory {
 /**
  * Type for `Realm.setLogLevel`
  */
-export type LogArgs = {
+export type LogOptions = {
```
Comment (kraenhansen) [inline packages/realm/src/Logger.ts:149]: We tend to use "options" to describe the type of object passed as the sole argument to a function.

File: packages/realm/src/Logger.ts [hunk 2]
```diff
@@ -161,27 +161,29 @@ export type Logger = (level: NumericLogLevel, message: string) => void;
...
  * @since 12.0.0
  * @deprecated Will be removed in v13.0.0
  */
 export type LoggerCallback1 = (level: LogLevel, message: string) => void;
-export type LoggerCallbackArgs = {
+
+/**
+ * Represents an entry in the log.
+ */
+export type LogEntry = {
```
Comment (kraenhansen) [inline packages/realm/src/Logger.ts:174]: I believe "entry" is semantically more precise.

File: packages/realm/src/Logger.ts [hunk 3]
```diff
@@ -161,27 +161,29 @@ export type Logger = (level: NumericLogLevel, message: string) => void;
...
   level: LogLevel;
+  /** The message of the log entry. */
   message: string;
 };
 /**
- * A callback passed to `Realm.setLogger`. Arguments are passed as a POJO.
- *
- * @param category   - The category (origin) of the log entry.
- * @param level      - The level of the log entry.
- * @param message    - The message of the log entry.
```
Comment (kraenhansen) [inline packages/realm/src/Logger.ts:181]: As these are not separate arguments to the callback, I've moved these descriptions into the properties on the object's type above.

File: packages/realm/src/Logger.ts (hunk 2 above)
Comment (kneth) [reply packages/realm/src/Logger.ts:174]: Yes, it reads better

**Review by kneth — APPROVED** [review]

**Pull request merged**